haskell/private/haskell_impl.bzl

```diff
@@ -393,6 +392,22 @@ def haskell_library_impl(ctx):
     srcs_files, import_dir_map = _prepare_srcs(ctx.attr.srcs)
     module_map = determine_module_names(srcs_files)
 
+    package_name = getattr(ctx.attr, "package_name", None)
+    version = getattr(ctx.attr, "version", None)
+    my_pkg_id = pkg_id.new(ctx.label, package_name, version)
+
+    # Determine file directories.
+    if my_pkg_id:
```

Comment:
This will never be `None`, so this condition is redundant.

Reply:
It can be empty, in which case the `else` branch will be entered.

Reply:
But, how? IIUC `my_pkg_id` is constructured by [`pkg_id.new` here](https://github.com/tweag/rules_haskell/blob/9efbbbecbc37c7c2fb72058f018da90a16621417/haskell/private/pkg_id.bzl#L30-L51), which alwasy returns a `struct` with fields like `label` and the like.

Reply:
I removed the if-then-else. I was confused on the context of this code.

Reply:
Thanks!